pkg/internal/util/cli/spinner.go

```diff
@@ -103,17 +123,37 @@ func (s *Spinner) Start() {
 
 // Stop signals the spinner to stop
 func (s *Spinner) Stop() {
+	// if there's nothing to stop, return early
+	if atomic.CompareAndSwapInt32(&s.running, 0, 0) {
+		return
+	}
+	// otherwise, send the stop signal, and wait for the reply
 	s.stop <- struct{}{}
+	<-s.stopped
 }
 
 // Write implements io.Writer, interrupting the spinner and writing to
 // the inner writer
 func (s *Spinner) Write(p []byte) (n int, err error) {
```

Comment:
previously this could trigger starting the spinner when it had never been started, which would occasionally add one of the early frames to the end of the CLI output.

that should not be possible now, we track if the spinner is running and have more robust synchronization

Reply:
in a further simplification: we don't actually need to stop the spinner, just hold the write lock, since it's now on the same object..